docrawl/docrawl_core.py

```diff
@@ -79,39 +79,21 @@ class DocrawlSpider(scrapy.spiders.CrawlSpider):
     def __init__(self, *a, **kw):
         self.docrawl_client = kw['docrawl_client']
 
-        self.kv_redis_key_screenshot = self.docrawl_client.kv_redis_keys.get('screenshot', 'screenshot')
+        self.kv_redis_key_screenshot = self.docrawl_client.kv_redis_keys.get(
+            'screenshot', 'screenshot'
+        )
         self.kv_redis_key_elements = self.docrawl_client.kv_redis_keys.get('elements', 'elements')
 
         self.browser = self._initialise_browser()
-        browser_meta_data = self.docrawl_client.get_browser_meta_data()
-        browser_meta_data['browser']['pid'] = self._determine_browser_pid()
 
-        self.docrawl_client.set_browser_meta_data(browser_meta_data)
-        self.screenshot_thread = None #needs to be initialized to None before execution
+        self.screenshot_thread = None  # needs to be initialized to None before execution
         self.start_requests()
 
     def _initialise_browser(self):
         browser_meta_data = self.docrawl_client.get_browser_meta_data()
-
-        docrawl_logger.info(f'Browser settings: {browser_meta_data}')
-
-        try:
-            self.driver_type = browser_meta_data['browser']['driver']
-        except Exception as e:
-            docrawl_logger.error(f'Error while loading driver type information: {e}')
-            self.driver_type = 'Firefox'
-
-        try:
-            self.headless = browser_meta_data['browser']['headless']
-        except Exception as e:
-            docrawl_logger.error(f'Error while loading headless mode information: {e}')
-            self.headless = False
-
-        try:
-            proxy_info = browser_meta_data['browser']['proxy']
-        except Exception as e:
-            docrawl_logger.warning(f'Error while loading proxy information: {e}')
-            proxy_info = None
+        self.driver_type = browser_meta_data['browser']['driver']
+        self.headless = browser_meta_data['browser']['headless']
+        proxy_info = browser_meta_data['browser']['proxy']
```

Comment:
I am worried here, there were most probably reasons why these three clauses were wrapped in try, except -> will need to get handled somewhere

Reply:
There is a possibility that some of the browser options are not passed / not correct. In such case we should still allow browser to start (even though we're basically allowing silent crash). Try/except can be replaced with .get() and default value, however I prefer to have try/except since it adds print message. Might make sense to replace general Exception with KeyError.

Reply:
Imo, it's a similar situation as with the `KeyError` above - if this data is not set when `docrawl_core.py` starts up, we have a programmatic error somewhere. Yes, we can default to some values, but we're silencing the error by doing so. Let's just make sure the error never happens instead?

I'll leave it to your judgement.